agimus_controller/agimus_controller/warm_start_base.py

```diff
@@ -0,0 +1,21 @@
+from __future__ import annotations
+import numpy as np
+from abc import ABC, abstractmethod
+from agimus_controller.ocp_base import OCPBase
+from agimus_controller.trajectory import TrajectoryPoint
+
+
+class WarmStartBase(ABC):
+    def __init__(self) -> None:
+        super().__init__()
+
+    @abstractmethod
+    def setup(self, ocp: OCPBase = None) -> None:
```

Comment:
Good question...
I am not sure how to proceed...
We could remove the setup function all together because it's gonna be used from outside here.
For the generation... It maybe enough to have a list of trajectory point for:
- "from_gravity_compensation"
- "from_previous_solution"
- ... from a topic?

Reply:
actually the Trajectory point refers also to the environment.
You can consider a "end_effector" as an "operational_point" and basiccaly saying that you can send any SE3 localisation of any objects in the room as well as the end-effector pose.

Reply:
I initially wanted to answer that I don't think this is a good idea for these reasons: 
- sending the full description of the environment for each point of the trajectory 
- the diffusion-based warm start call is likely gonna be slower that 100Hz so we probably might want to receive its results in a node call back.   

On the other hand, Arthur's collision MPC requires a reference for the obstacles position, and potentially full description if we want to be really reactive. We did not discuss where this might come from: another OCP/MPC `update_obstacles` member function called before run? Not sure.

Reply:
we can simply have the diffusion model as a ROS class. because it's gonna be built in the ROS node anyway.
Have the basic api and sync the environment or diffusion model through callbacks.
This does not prevent to use the current api.